### PR TITLE
doc: add v1.4.2 to master

### PR DESCRIPTION
**Pull request by greg-fer**
Updated master with v1.4.2 files.
Updated v1.4.1 known issues with v1.4.2 tag.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

- [x] Update release notes with the final version
- [x] Update known issues with the final version
- [x] Edit nrfxlib version to 1.4.2

### CODE REVIEW

**Comment by greg-fer** [comment]
Pending updates from #3637

File: doc/nrf/links.txt [hunk 1]
```diff
@@ -95,6 +95,7 @@
 
 .. _`known issues page on the master branch`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/known_issues.html
 .. _`known issues for nRF Connect SDK v1.4.99-dev1`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/known_issues.html?v=v1-4-99-dev1
+.. _`known issues for nRF Connect SDK v1.4.2`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/known_issues.html?v=v1-4-1
```
Comment (ru-fu) [inline doc/nrf/links.txt:98]:
```suggestion
.. _`known issues for nRF Connect SDK v1.4.2`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/known_issues.html?v=v1-4-2
```

Comment (greg-fer) [reply doc/nrf/links.txt]: :/ Thanks!

**Comment by greg-fer** [comment]
Removing DNM on request from @shantha-14 . @tejlmand , @ru-fu , please review when you have a second.

File: doc/nrf/known_issues.rst [hunk 2]
```diff
@@ -81,13 +82,25 @@ NCSDK-5574: Warnings during FOTA
...
 .. rst-class:: v1-0-0 v0-4-0 v0-3-0
 
 Sending data before connecting to nRF Cloud
-  The :ref:`asset_tracker` sample does not wait for connection to nRF Cloud before trying to send data.
-  This causes the sample to crash if the user toggles one of the switches before the board is connected to the cloud.
+  The :ref:`asset_tracker` application does not wait for connection to nRF Cloud before trying to send data.
+  This causes the application to crash if the user toggles one of the switches before the board is connected to the cloud.
+
+.. rst-class:: v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0 v1-1-0 v1-0-0 v0-4-0 v0-3-0
+
```
Comment (ru-fu) [inline doc/nrf/known_issues.rst:94]: Two blank lines - I don't think this will cause issues, but better to fix.

Comment (greg-fer) [reply doc/nrf/known_issues.rst:94]: Yes, fixed now.

**Review by ru-fu — APPROVED** [review]

**Comment by greg-fer** [comment]
Updated nrfxlib version to v1.4.2 to match the last-minute change in #3652 .

**Review by tejlmand — APPROVED** [review]

**Pull request merged**